streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -53,6 +55,76 @@ public final class TaskAssignmentUtils {
 
     private TaskAssignmentUtils() {}
 
+    /**
+     * Return an {@code AssignmentError} for a task assignment created for an application.
+     *
+     * @param applicationState The application for which this task assignment is being assessed.
+     * @param taskAssignment   The task assignment that will be validated.
+     *
+     * @return {@code AssignmentError.NONE} if the assignment created for this application is valid,
+     *         or another {@code AssignmentError} otherwise.
+     */
+    public static AssignmentError validateTaskAssignment(final ApplicationState applicationState,
```

Comment:
just to save myself some review time: was anything changed here or is it just a copy-paste from SPA to here?

Reply:
Copy paste, no changes at all.